### PR TITLE
Simple crop models

### PR DESCRIPTION
**Pull request by HamishBrownPFR**
this pull request contains the content originally put up on #8479.  I have closed that because the branch became cluttered with other work and put this up to reduce the amount of stuff contained in this PR to that originally intended.

Working on #8175 #7869 #7075 (suite of simple crop, tree and pasture models)

Added fRadnAllZones variable into ICanopy. This is the proportion of the total radiation over both zones that a crops in a specific zone is intercepting. We have been calculating this in manager scripts but seems safer to do it in Microclimate

Some harmonizing of approaches and functionality between SCRUM, STRUM and SPRUM

### CODE REVIEW

**Comment by HamishBrownPFR** [comment]
@hol353 @par456 @ric394 

This is a tidy up of https://github.com/APSIMInitiative/ApsimX/pull/8479.  The comments on that had been responded to but the branch needed to be updated before it could be merged.  Unfortunately, I had committed so other stuff onto the same branch in the time that lapsed so could not update.  I have set up a clean branch and pull request.  This should be good to merge now.

If you are happy this is good to merge please do so.

**Comment by HamishBrownPFR** [comment]
@hol353 could you take a look at this one when you have a min

File: Models/Interfaces/ICanopy.cs [hunk 1]
```diff
@@ -48,6 +48,9 @@ public interface ICanopy
 
         /// <summary>Sets the light profile.</summary>
         CanopyEnergyBalanceInterceptionlayerType[] LightProfile { set; }
+
+        /// <summary>The fraction of total radiatin over all zones intercepted by this canopy</summary>
+        double fRadnAllZones { get; set; }
```
Comment (hol353) [inline Models/Interfaces/ICanopy.cs:53]: I don't like fRadnAllZones in ICanopy. None of the classes that implement ICanopy use this so it shouldn't be in the interface. MicroClimateZone sets it so I think it should be in here. MicroClimateZone could have a method to provide the value to a manager script for a given canopy. e.g.

```c#
Dictionary<ICanopy, double> fRadnAllZones;
public double FractionRadnAllZones(ICanopy canopy)
{
    return fRadnAllZones[canopy];
}
```

File: Models/PMF/SimplePlantModels/SprumPastureInstance.cs [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+            phenology.SetAge(AgeAtSimulationStart);
+            summary.WriteMessage(this,"Some of the message above is not relevent as SPRUM has no notion of population, bud number or row spacing." +
+                " Additional info that may be useful.  " + this.Name + " is established "
+                ,MessageType.Information); 
+        }
+
+        /// <summary>
+        /// Data structure that holds STRUM parameter names and the cultivar overwrite they map to
+        /// </summary>
+        public Cultivar coeffCalc()
```
Comment (hol353) [inline Models/PMF/SimplePlantModels/SprumPastureInstance.cs:269]: Method names start with upper case.

File: Models/PMF/SimplePlantModels/StrumTreeInstance.cs [hunk 3]
```diff
@@ -52,37 +52,41 @@ public class StrumTreeInstance: Model
         [Description("End of Leaf fall (Days after Winter Solstice)")]
         public int EndLeafFallDAWS { get; set; }
 
+        /// <summary>Grow roots into neighbouring zone (yes or no)</summary>
+        [Separator("Tree Dimnesions")]
+        [Description("Grow roots into neighbouring zone (yes or no)")]
+        public bool RootThyNeighbour { get; set; }
```
Comment (hol353) [inline Models/PMF/SimplePlantModels/StrumTreeInstance.cs:58]: Yeah, no. Very creative and very Hamish like but we need a better name.

**Review by hol353 — CHANGES_REQUESTED** [review]

File: Models/Interfaces/ICanopy.cs (hunk 1 above)
Comment (HamishBrownPFR) [reply Models/Interfaces/ICanopy.cs:53]: This was a bit handy for reporting but the core reason why I added it I moved away from so it was no longer necessary.  I have removed this new variable from the current push

File: Models/PMF/SimplePlantModels/SprumPastureInstance.cs (hunk 2 above)
Comment (HamishBrownPFR) [reply Models/PMF/SimplePlantModels/SprumPastureInstance.cs:269]: changed in current push

File: Models/PMF/SimplePlantModels/StrumTreeInstance.cs (hunk 3 above)
Comment (HamishBrownPFR) [reply Models/PMF/SimplePlantModels/StrumTreeInstance.cs:58]: Changed in current push

**Comment by hol353** [comment]
This branch (HamishBrownPFR/SimpleCropModels) doesn't seem to have fRadnAllZones  removed but I notice your PFR/SimpleCropModels branch does. Do you need to pull from PFR/SimpleCropModels?

**Comment by HamishBrownPFR** [comment]
Thanks Dean.  To many branches.  Hoping to get some closed soon.

File: Models/Surface/SurfOrganicMatterType.cs [hunk 4]
```diff
@@ -106,6 +107,10 @@ public class ResidueCanopy : ICanopy
         /// <summary>Sets the actual water demand.</summary>
         public double WaterDemand { get; set; }
 
+        /// <summary>The fraction of total radiatin over all zones intercepted by this canopy</summary>
+        [Units("0-1")]
+        public double fRadnAllZones { get; set; }
```
Comment (hol353) [inline Models/Surface/SurfOrganicMatterType.cs:112]: Remove this one as well.

**Review by hol353 — CHANGES_REQUESTED** [review]

File: Models/PMF/Organs/SimpleLeaf.cs [hunk 5]
```diff
@@ -331,6 +331,10 @@ public SimpleLeaf()
         [Units("mm")]
         public double WaterDemand { get; set; }
 
+        /// <summary>The fraction of total radiatin over all zones intercepted by this canopy</summary>
+        [Units("0-1")]
+        public double fRadnAllZones { get; set; }
```
[the remark this answers is not in the document]
Comment (HamishBrownPFR) [reply Models/PMF/Organs/SimpleLeaf.cs]: I thought I had.  Done now

File: Models/Surface/SurfOrganicMatterType.cs (hunk 4 above)
Comment (HamishBrownPFR) [reply Models/Surface/SurfOrganicMatterType.cs]: Dito, there were a couple of others too.

**Review by hol353 — APPROVED** [review]

**Pull request merged**